pynocular/backends/base.py

```diff
@@ -0,0 +1,137 @@
+"""Contains base classes for defining database backends"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Set
+
+from sqlalchemy import Column, Table
+from sqlalchemy.sql.elements import BinaryExpression, UnaryExpression
```

Comment:
Yeah, I kept the SQLAlchemy part from v1 in place at the DatabaseModel layer since it has the table and column objects. When we go write the UDD backend, we can translate the table and column objects to whatever API calls we need. We most likely won't use any of the query building parts of SQLAlchemy.